presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -246,24 +246,31 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
                 bucketFilter);
     }
 
-    private Optional<HiveBucketHandle> getBucketHandle(Table table, ConnectorSession session)
+    private Optional<HiveBucketHandle> getBucketHandle(
+            Table table,
+            ConnectorSession session,
+            TupleDomain<ColumnHandle> effectivePredicate)
     {
         // never ignore table bucketing for temporary tables as those are created such explicitly by the engine request
         if (table.getTableType().equals(TEMPORARY_TABLE)) {
             return getHiveBucketHandle(table);
         }
 
         Optional<HiveBucketHandle> hiveBucketHandle = getHiveBucketHandle(table);
-        if (!hiveBucketHandle.isPresent()) {
+        if (!hiveBucketHandle.isPresent() || shouldIgnoreTableBucketing(session)) {
             return Optional.empty();
         }
 
+        if (queryUsesHiveBucketColumn(effectivePredicate)) {
```

Comment:
wouldn't this still return Optional.empty() if shouldIgnoreTableBucketing is true, since that condition is checked first?

Reply:
Yes, this scenario is covered by `TestHiveIntegrationSmokeTest::testIgnoreTableBucketing`. Query will fail if it uses bucket column when shouldIgnoreTableBucketing is true.

Reply:
sorry, I misunderstood the purpose of this pr.  I thought this PR was so that it would succeed in that case, but now I see it's only for when the table bucket count is small.

Reply:
You are right, the purpose of this PR is to not ignore table bucketing when bucket count is small and shouldIgnoreTableBucketing is false. 
There are two configs `hive.ignore-table-bucketing` and `hive.min-bucket-count-to-not-ignore-table-bucketing`.  If `hive.ignore-table-bucketing` is true, tables are always ignored. Otherwise (like in this PR), if depends on `hive.min-bucket-count-to-not-ignore-table-bucketing` and whether bucket column is being used.